### PR TITLE
#14967. Send full name instead alias when attach contact in MegaChat

### PR DESCRIPTION
**Pull request by jgandres**

### CODE REVIEW

**Comment by sergiohs84** [comment]
trigger compilation

File: examples/qtmegachatapi/MegaChatApplication.cpp [hunk 1]
```diff
@@ -680,21 +680,31 @@ void MegaChatApplication::onRequestFinish(MegaChatApi *, MegaChatRequest *reques
...
              MegaChatHandle userHandle = request->getUserHandle();
              int errorCode = error;
+             mFirstnameFetching[userHandle] = false;
              if (errorCode == MegaChatError::ERROR_OK)
              {
                 const char *firstname = request->getText();
                 if ((strlen(firstname)) == 0)
                 {
-                    this->mMegaChatApi->getUserEmail(userHandle);
+                    auto it = mFirstnamesMap.find(userHandle);
```
Comment (alber2510) [inline examples/qtmegachatapi/MegaChatApplication.cpp]: It's not necessary `find` in a map before `erase`. `erase` from a map with a non-existent key is safe. The code is correct. You can remove if you want or keep in this way

File: examples/qtmegachatapi/MegaChatApplication.cpp [hunk 2]
```diff
@@ -680,21 +680,31 @@ void MegaChatApplication::onRequestFinish(MegaChatApi *, MegaChatRequest *reques
...
                     break;
                 }
                 mFirstnamesMap[userHandle] = firstname;
-                mFirstnameFetching[userHandle] = false;
                 mMainWin->updateContactTitle(userHandle,firstname);
                 mMainWin->updateMessageFirstname(userHandle,firstname);
              }
              else if (errorCode == MegaChatError::ERROR_NOENT)
              {
+                auto it = mFirstnamesMap.find(userHandle);
```
Comment (alber2510) [inline examples/qtmegachatapi/MegaChatApplication.cpp]: It's not necessary `find` in a map before `erase`. `erase` from a map with a non-existent key is safe. The code is correct. You can remove if you want or keep in this way

**Comment by jgandres** [comment]
trigger compilation

**Pull request merged**